status/pcpp/evaluator.py

```diff
@@ -0,0 +1,147 @@
+from pcpp.preprocessor import STRING_TYPES
+
+# The width of signed integer which this evaluator will use
+INTMAXBITS = 64
+
+# Some Python 3 compatibility shims
+if sys.version_info.major < 3:
+    INTBASETYPE = long
+else:
+    INTBASETYPE = int
+
+class Int(INTBASETYPE):
+    """A signed integer within a preprocessor expression, bounded
+    to within INT_MIN and INT_MAX. Overflow is handled like a CPU,
+    despite being UB, as that's what GCC and clang do.
+    
+    >>> Int(5)
+    Int(5)
+    >>> Int(5) * Int(2)
+    Int(10)
+    >>> Int(5) * 2
+    Int(10)
+    >>> Int(50) % 8
+    Int(2)
+    >>> -Int(5)
+    Int(-5)
+    >>> +Int(-5)
+    Int(-5)
+    >>> ~Int(5)
+    Int(-6)
+    >>> Int(6) & 2
+    Int(2)
+    >>> Int(4) | 2
+    Int(6)
+    >>> Int(6) ^ 2
+    Int(4)
+    >>> Int(2) << 2
+    Int(8)
+    >>> Int(8) >> 2
+    Int(2)
+    >>> Int(9223372036854775808)
+    Int(-9223372036854775808)
+    >>> Int(-9223372036854775809)
+    Int(9223372036854775807)
+    >>> Int(18446744073709551615)
+    Int(-1)
+    """
+    INT_MIN = -(1 << (INTMAXBITS - 1))
+    INT_MAX = (1 << (INTMAXBITS - 1)) - 1
+    INT_MASK = (1 << INTMAXBITS) - 1
+    @classmethod
+    def __clamp(cls, value):
+        return ((value - cls.INT_MIN) & cls.INT_MASK) + cls.INT_MIN
+    def __new__(cls, value, *args, **kwargs):
+        return super(Int, cls).__new__(cls, cls.__clamp(value))
+    def __add__(self, other):
+        return self.__class__(self.__clamp(super(Int, self).__add__(other)))
+    def __sub__(self, other):
+        return self.__class__(self.__clamp(super(Int, self).__sub__(other)))
+    def __mul__(self, other):
+        return self.__class__(self.__clamp(super(Int, self).__mul__(other)))
+    def __div__(self, other):
+        return self.__class__(self.__clamp(super(Int, self).__div__(other)))
+    def __mod__(self, other):
+        return self.__class__(self.__clamp(super(Int, self).__mod__(other)))
+    def __neg__(self):
+        return self.__class__(self.__clamp(super(Int, self).__neg__()))
+    def __invert__(self):
+        return self.__class__(self.__clamp(super(Int, self).__invert__()))
+    def __and__(self, other):
+        return self.__class__(self.__clamp(super(Int, self).__and__(other)))
+    def __or__(self, other):
+        return self.__class__(self.__clamp(super(Int, self).__or__(other)))
+    def __pos__(self):
+        return self.__class__(self.__clamp(super(Int, self).__pos__()))
+    def __pow__(self, other):
+        return self.__class__(self.__clamp(super(Int, self).__pow__(other)))
+    def __lshift__(self, other):
+        return self.__class__(self.__clamp(super(Int, self).__lshift__(other)))
+    def __rshift__(self, other):
+        return self.__class__(self.__clamp(super(Int, self).__rshift__(other)))
+    def __xor__(self, other):
+        return self.__class__(self.__clamp(super(Int, self).__xor__(other)))
+    def __repr__(self):
+        return "Int(%d)" % INTBASETYPE(self)
+
+#def execute_expr(token):
```

Comment:
Please don't commit commented code. If you want to keep it, commit with it and then remove in next commit. So you can go back if ever required

Reply:
I'm not modifying the embedded pcpp.